macros/src/trans.rs

```diff
@@ -204,6 +213,46 @@ fn init(app: &App, main: &mut Vec<Tokens>, root: &mut Vec<Tokens>) {
         exprs.push(quote!(init::Resources::new()));
     }
 
+    let mut late_resources = vec![];
+    let has_late_resources = app.resources.iter()
```

Comment:
Same as above: this could use `any`

Reply:
Actually it just occurred to me that this could be done in a single pass over the resources by collecting the "early" resources and the "late" resources in vectors and then doing the appropriate codegen depending on whether the vector is empty or not.

Not sure which one would be easier to read.